test/browser/anchor-manager.test.jsx

```diff
@@ -0,0 +1,261 @@
+    sinon.stub(console, 'warn');
+
+    manager.register(fakeAnchor('foo'));
+    manager.unregister(fakeAnchor('foo'));
+    console.warn.callCount.should.equal(1);
+    console.warn.getCall(0).args.should.eql([
+      'Anchor mismatch for #foo.'
+    ]);
+    console.warn.restore();
+  });
+
+  it('should no-op if hash changes between unmatched anchors', function() {
+    var anchor = manager.register(fakeAnchor('foo'));
+    manager.simulateHashChange('blah');
+    anchor.handleNavigateFromAnchor.callCount.should.equal(0);
+    anchor.handleNavigateToAnchor.callCount.should.equal(0);
+    should(manager.currentAnchor).be.null;
+  });
+
+  it('should no-op if hash matches current anchor', function() {
+    var anchor = manager.register(fakeAnchor('foo'));
+    manager.currentAnchor = anchor;
+    manager.simulateHashChange('foo');
+    anchor.handleNavigateFromAnchor.callCount.should.equal(0);
+    anchor.handleNavigateToAnchor.callCount.should.equal(0);
+    manager.currentAnchor.should.equal(anchor);
+  });
+
+  it('should navigate from unmatched anchor to matched anchor', function() {
+    var anchor = manager.register(fakeAnchor('foo'));
+    manager.simulateHashChange('foo');
+    anchor.handleNavigateFromAnchor.callCount.should.equal(0);
+    anchor.handleNavigateToAnchor.callCount.should.equal(1);
+    manager.currentAnchor.should.equal(anchor);
+  });
+
+  it('should navigate from old anchor to unmatched anchor', function() {
+    var fooAnchor = manager.register(fakeAnchor('foo'));
+    manager.currentAnchor = fooAnchor;
+    manager.simulateHashChange('blah');
+    fooAnchor.handleNavigateFromAnchor.callCount.should.equal(1);
+    fooAnchor.handleNavigateToAnchor.callCount.should.equal(0);
+    should(manager.currentAnchor).be.null;
+  });
+
+  it('should navigate from old anchor to matched anchor', function() {
+    var fooAnchor = manager.register(fakeAnchor('foo'));
+    var barAnchor = manager.register(fakeAnchor('bar'));
+    manager.currentAnchor = fooAnchor;
+    manager.simulateHashChange('bar');
+    fooAnchor.handleNavigateFromAnchor.callCount.should.equal(1);
+    fooAnchor.handleNavigateToAnchor.callCount.should.equal(0);
+    barAnchor.handleNavigateFromAnchor.callCount.should.equal(0);
+    barAnchor.handleNavigateToAnchor.callCount.should.equal(1);
+    manager.currentAnchor.should.equal(barAnchor);
+  });
+});
+
+describe('AnchorManagerMixin', function() {
+  var manager, anchors, clock;
+
+  var SampleAnchorClass = React.createClass({
+    mixins: [AnchorManagerMixin],
+    render: function() {
+      return <div></div>;
+    }
+  });
+
+  var renderAnchor = function(props) {
+    var anchor = React.addons.TestUtils.renderIntoDocument(
+      <SampleAnchorClass anchorManager={manager} {...props}/>
+    );
+    anchors.push(anchor);
+    return anchor;
+  };
+
+  var unmountAnchor = function(anchor) {
+    React.unmountComponentAtNode(anchor.getDOMNode().parentNode);
+    anchors.splice(anchors.indexOf(anchor, 1));
```

Comment:
Oh shoot, this might be a typo--I just wanted to remove the anchor from the list... I think I meant to write this:

```js
anchors.splice(anchors.indexOf(anchor), 1);
```

eep